gateway/ln-gateway/src/actor_fork.rs

```diff
@@ -42,7 +53,110 @@ impl GatewayActor {
+                    Ok(None) => {
+                        warn!("HTLC stream closed by service");
+                        None
+                    }
+                    Err(e) => {
+                        error!("HTLC stream closed with error: {:?}", e);
+                        None
+                    }
+                } {
+                    if subscription.is_shutting_down() {
+                        info!("Shutting down HTLC subscription");
+                        break;
+                    }
+
+                    // TODO: Assert short channel id matches the one we subscribed to, or cancel processing of intercepted HTLC
+                    // TODO: Assert the offered fee derived from invoice amount and outgoing amount is acceptable or cancel processing of intercepted HTLC
+                    // TODO: Assert the HTLC expiry or cancel processing of intercepted HTLC
+
+                    let hash = match sha256::Hash::from_slice(&payment_hash) {
+                        Ok(hash) => hash,
+                        Err(e) => {
+                            let fail = "Failed to parse payment hash";
+
+                            error!("{}: {:?}", fail, e);
+                            let _ = lnrpc
+                                .complete_htlc(CompleteHtlcsRequest {
+                                    intercepted_htlc_id,
+                                    action: Some(Action::Cancel(Cancel {
+                                        reason: fail.to_string(),
+                                    })),
+                                })
+                                .await;
+                            continue;
+                        }
+                    };
+
+                    let amount_msat = Amount::from_msats(outgoing_amount_msat);
+
+                    match actor.buy_preimage_internal(&hash, &amount_msat).await {
+                        Ok(preimage) => {
+                            info!("Successfully processed intercepted HTLC");
+                            if let Err(e) = lnrpc
+                                .complete_htlc(CompleteHtlcsRequest {
+                                    intercepted_htlc_id,
+                                    action: Some(Action::Settle(Settle {
+                                        preimage: preimage.0.to_vec(),
+                                    })),
+                                })
+                                .await
+                            {
+                                error!("Failed to complete HTLC: {:?}", e);
+                                // Note: To prevent loss of funds for the gateway,
+                                // we should either retry completing the htlc or reclaim funds from the federation
+                            };
+                        }
+                        Err(e) => {
+                            error!("Failed to process intercepted HTLC: {:?}", e);
+                            let _ = lnrpc
```

Comment:
why are we ignoring the `Result` returned by `complete_htlc()`?

Reply:
this specific complete htlc  requires no futher action. If we fail to send the complete htlc message, lightning node will still cancel HTCL after expiry period lapses. Result can be safely ignored.

Added a note and TODO comment

Reply:
ah, thanks. :+1:  follow-up: this expiry time is usually rather short, correct?  Does the expiry correspond to [this field](https://github.com/fedimint/fedimint/blob/cd413f1598b4ee209a19a9f3ba8db59cf0828be8/gateway/ln-gateway/proto/gatewaylnrpc.proto#L81)?

Reply:
Yes.
We'll need to dig into expiry a little more down the road for optimizations of intercepted htlc processing.
For example, the gateway might want to reject / ignore a htlc in cases where that field indicates the htlc is super close to expiry...